### PR TITLE
Rename to GPUShaderModule.getCompilationInfo()

### PR DESCRIPTION
**Pull request by takahirox**
To follow the upcoming WebGPU spec change

https://github.com/gpuweb/gpuweb/pull/3805#issuecomment-1433931903

Requires
* https://github.com/gpuweb/gpuweb/pull/3805 needs to be merged
* [@webgpu/types](https://github.com/gpuweb/types) needs to updated with the above change

so this PR is still draft. I will switch to ready for review when the aboves are updated.

Issue: #2290

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [x] Tests are properly located in the test tree.
- [x] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [x] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [x] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Review by Kangz — APPROVED** [review]
LGTM once the gpuweb change is landed and gpuweb/types is rolled. Or you could make a gpuweb/types update too.

**Comment by takahirox** [comment]
> Or you could make a gpuweb/types update too.

I want to wait for a week for WebGPU spec and types updates. If they don't land by then, I will make a PR to types for the change that goes ahead by the spec.

**Review by kainino0x — APPROVED** [review]
I just saw the comment on https://chromium-review.googlesource.com/c/chromium/src/+/4267951 about landing this CTS change before landing the Chrome change so Chrome doesn't have to have a deprecation period. We'll have a short gap in coverage but this is fine with me, LGTM.

**Pull request merged**
